plenum/test/pool_transactions/test_txn_pool_manager.py

```diff
@@ -19,6 +24,33 @@ def pool_node_txns(poolTxnData):
     return node_txns
 
 
+def test_twice_demoted_node_dont_write_txns(txnPoolNodeSet,
+                                            looper, sdk_wallet_stewards, sdk_pool_handle):
+    request_count = 5
+    demoted_node = txnPoolNodeSet[2]
+
+    def get_node_prods_count(node):
+        return node.metrics._accumulators[MetricsName.NODE_PROD_TIME].count
+
+    def is_prods_run(node, old, diff):
+        new = get_node_prods_count(node)
+        assert old + diff <= new
+
+    demote_node(looper, sdk_wallet_stewards[2], sdk_pool_handle, demoted_node)
```

Comment:
please add check, that node was actually demoted for all other nodes.

Reply:
ok